File: frontend/src/app/models/response/company-response.ts
import { EBusinessType } from "../enum/ebusiness-type";
import { ECompanySize } from "../enum/ecompany-size";
import { EIndustry } from "../enum/eindustry";
import { ContactResponse } from "./contact-response";
import { UserResponse } from "./user-response";

export interface CompanyResponse {
    id: string;
    name: string;
    imageUrl: string;
    size: ECompanySize;
    industry: EIndustry;
    businessType: EBusinessType;
    country: string;
    city: string;
    address: string;
    website: string;
    contacts: ContactResponse[];
    salesOwner: UserResponse;
    createdAt: Date;
    updatedAt: Date;
    createdBy: string;
    updatedBy: string;
}

File: backend/src/main/java/org/gestionstock/stock/Payload/Response/CompanyResponse.java
package org.gestionstock.stock.Payload.Response;

import java.util.List;

import org.gestionstock.stock.Enum.EBusinessType;
import org.gestionstock.stock.Enum.ECompanySize;
import org.gestionstock.stock.Enum.EIndustry;

public record CompanyResponse(
    String id,
    String name,
    String imageUrl,
    ECompanySize size,
    EIndustry industry,
    EBusinessType businessType,
    String country,
    String city,
    String address,
    String website,
    List<ContactResponse> contacts,
    UserResponse salesOwner,
    String createdAt,
    String updatedAt,
    String createdBy,
    String updatedBy
) {
    
}
